### ISSUE TITLE
PromiseProxy calling success handler with a rejected promise

### ISSUE THREAD

**Issue by deepan83** [emberjs/ember.js#15694]
I've noticed that in Ember 2.15, the PromiseProxy now calls the success handler of the promise with a rejected promise setting. Its ok in 2.14.x and this is causing a lot of issues in our apps. 

```
App.ApplicationController = Ember.Controller.extend({
  init() {
    let ObjectPromiseProxy = Ember.ObjectProxy.extend(Ember.PromiseProxyMixin);

    ObjectPromiseProxy
      .create({promise: Ember.RSVP.reject()})
      .then(() => {
        alert('Unexpected');
       })
       .catch(() => {
         alert('Expected');
       })
    
   }
});

```

https://jsbin.com/vejosugiso/edit?html,js,output

**Comment by bekzod** [comment]
can confirm fixed in `rsvp 4.0.2` https://github.com/tildeio/rsvp.js/pull/501 
only `ember 2.15 affected`

**Comment by rwjblue** [comment]
Fix was just landed in https://github.com/emberjs/ember.js/pull/15685, and will be backported for 2.15.2.

**Comment by bekzod** [comment]
@rwjblue isn't this forgotten to be released with `2.15.3` :P ?

**Comment by deepan83** [comment]
Still seems to be an issue with 2.16.x
